src/utils/utils.ts

```diff
@@ -1117,3 +1118,51 @@ export const validConcentrationMidPriceTick = (
 
   return midPriceTick
 }
+
+type LiquidityTickWithoutIndex = Omit<LiquidityTick, 'index'>
+
+export const getLiquidityTicksByPositionsList = (
+  poolKey: PoolKey,
+  positions: Position[]
+): LiquidityTick[] => {
+  const ticks: Record<number, LiquidityTickWithoutIndex> = {}
+
+  positions.forEach(position => {
+    if (poolKeyToString(position.poolKey) === poolKeyToString(poolKey)) {
+      updateTick(ticks, Number(position.lowerTickIndex), position.liquidity, true)
+      updateTick(ticks, Number(position.upperTickIndex), position.liquidity, false)
+    }
+  })
+
+  return Object.entries(ticks).map(([index, { liquidityChange, sign }]) => ({
+    index: BigInt(index),
+    liquidityChange,
+    sign
+  }))
+}
+
+export const updateTick = (
+  ticks: Record<number, LiquidityTickWithoutIndex>,
+  tickIndex: number,
+  positionLiquidity: bigint,
+  isLowerTick: boolean
+) => {
+  if (!ticks[tickIndex]) {
+    ticks[tickIndex] = {
+      liquidityChange: positionLiquidity,
+      sign: isLowerTick
+    }
+  }
+
+  const tick = ticks[tickIndex]
+
+  if (tick.sign === isLowerTick) {
+    tick.liquidityChange += positionLiquidity
```

Comment:
What if previously the sign on this tick was `-1/false`, because there is upper tick of the position. The next position you check, start with the same tick as the previous one ended - in this case there is possible to change sign from `-1/false` to `1/true`.

So IMO you should handle `sign` also here